src/error.rs

```diff
@@ -114,7 +116,19 @@ pub struct Error {
     /// A human-readable error message, usually a sentence explaining what went wrong.
     pub message: String,
     /// The http status code
-    pub status_code: http::StatusCode,
+    pub status_code: StatusCode,
+}
+
+impl EndpointError for Error {
+    fn try_from_response(
+        response: ruma_api::exports::http::Response<Vec<u8>>,
+    ) -> Result<Self, ResponseDeserializationError> {
+        if let Ok(error_body) = from_slice::<ErrorBody>(response.body()) {
```

Comment:
Style nit: Since `from_slice` is a very generic name and it's not immediately obvious from context, please don't import it into scope and instead refer to it with its full path.

```suggestion
        if let Ok(error_body) = serde_json::from_slice::<ErrorBody>(response.body()) {
```

Reply:
Yea I looked around a bit and all I could find were uses of this in tests that imported it directly but I agree its too vague.